corpus_merge.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash -eu
+#
+# Merges new fuzzing inputs into an existing corpus, only keeping the inputs
+# that increase coverage as measured by Go's native fuzzing engine. Prefers
+# smaller inputs over larger ones.
+#
+# Ideally, Go's native fuzzing engine would provide merge functionality via a
+# command-line flag. But until https://github.com/golang/go/issues/49290 is
+# resolved, we'll have to use our own merging algorithm.
+#
+# Example usage to merge new inputs into the FuzzPong corpus:
+#   $ ./corpus_merge.sh lnwire/testdata/fuzz/FuzzPong \
+#       $(go env GOCACHE)/fuzz/github.com/lightningnetwork/lnd/lnwire/FuzzPong \
+#       ../lnd/lnwire FuzzPong
+
+if [[ "$#" -ne 4 ]]; then
+	echo "Usage: $0 DEST_CORPUS SRC_CORPUS PACKAGE_DIR FUZZ_TARGET_NAME"
+	exit 1
+fi
+
+readonly DEST_DIR="$1"
```

Comment:
Nice, wasn't even aware of the `readonly` feature in `bash`. Are we concerned about multi platform compatibility? While doing a quick search for this feature I saw a note that this might behave differently on some system, especially with POSIX mode (https://stackoverflow.com/questions/30362831/what-is-difference-in-declare-r-and-readonly-in-bash), though not sure if that is actually used by some operating systems?
I guess we should just make sure this script works on both Linux as well as MacOS, that should cover most of our devs.

Reply:
I don't have an easy way to test on MacOS, but the hashbang specifies `/bin/bash -eu`, and I'd expect that to avoid unexpected flags like POSIX mode.  It should also ensure `bash` is invoked instead of MacOS's `zsh`.

Of course, the hashbang doesn't help if the user runs the script manually (e.g. `bash --posix ./corpus_merge.sh ...`), but the user probably *should* know what they're doing in that case.